src/remote/workspace.py

```diff
@@ -235,3 +264,34 @@ def clear_remote(self) -> None:
     def create_remote(self) -> None:
         """Remove remote directory"""
         self.execute(f"mkdir -p {self.remote.directory}", simple=True)
+
+    @contextmanager
+    def hot_reload(self, hot_reload_time: float = 1) -> None:
+        """Sync local files whenever a change is made."""
+        has_changes = Event()
+        # Set up a worker thread to process the changes after the changes are settled as per the hot reload time.
+        worker = Thread(target=self.process_events, args=(has_changes, hot_reload_time))
+        # Start observing the local workspace.
+        observer = Observer()
+        observer.schedule(SyncedWorkSpaceHandler(has_changes), self.local_root, recursive=True)
+        try:
+            worker.start()
+            observer.start()
+            yield
+        finally:
+            observer.stop()
+            worker.do_run = False
+            observer.join()
+            worker.join()
+
+    def process_events(self, has_changes: Event, hot_reload_time: float = 1) -> None:
+        """Process file system events."""
+        thread = threading.currentThread()
+        # The do_run attribute helps in gracefully stopping the thread.
+        while getattr(thread, "do_run", True):
```

Comment:
Creating a class that has `Thread` as the base and a declared `do_run` attribute will make the logic much cleaner